fairroot/geobase/FairGeoSet.h

```diff
@@ -51,18 +51,24 @@ class FairGeoSet : public TNamed
         FairGeoCopyNode& operator=(const FairGeoCopyNode&);
     };
 
-    TList* volumes;         /** list of volumes */
-    TList* masterNodes;     /** pointer to list of mother volumes from other detector parts*/
-    Int_t maxSectors{0};    /** maximum number of sectors (-1 for detectors outside any sector)*/
-    Int_t maxKeepinVolumes; /** maximum number of keepin volumes per sector*/
-    Int_t maxModules;       /** maximum number of modules per sector*/
-    TArrayI* modules;       /** Module's array.*/
-    FairGeoShapes* pShapes; /** pointer to the class FairGeoShapes*/
-    TString geoFile;        /** name of geometry input file or Oracle*/
-    TString author;         /** author of the media version*/
-    TString description;    /** description of the version*/
+    TList* volumes{new TList()};     /** list of volumes */
+    TList* masterNodes{nullptr};     /** pointer to list of mother volumes from other detector parts*/
+    Int_t maxSectors{0};             /** maximum number of sectors (-1 for detectors outside any sector)*/
+    Int_t maxKeepinVolumes{0};       /** maximum number of keepin volumes per sector*/
+    Int_t maxModules{0};             /** maximum number of modules per sector*/
+    TArrayI* modules{nullptr};       /** Module's array.*/
+    FairGeoShapes* pShapes{nullptr}; /** pointer to the class FairGeoShapes*/
+    TString geoFile{""};             /** name of geometry input file or Oracle*/
+    TString author{""};              /** author of the media version*/
+    TString description{""};         /** description of the version*/
+
+    FairGeoSet() = default;
+    FairGeoSet(const char* name, Int_t maxMods, Int_t maxSecs = 0)
```

Comment:
Why don't use a std::string& ?

Reply:
Because `TNamed(name, …)` also only has a `const char*`. so we'd need to construct a full `std::string` just do call `.c_str()` on it. Don't know, if this is really useful.

We could go for a `std::string_view`. This would then be converted to a TString which would be copy-constructed into the TString in TNamed.

Reply:
> Because `TNamed(name, …)` also only has a `const char*`.

And this [`TNamed::TNamed(const TString&, const TString&)`](https://root.cern.ch/doc/master/classTNamed.html#a92b15a72a05ddf8f23c67023a29ee6a9) - so perhaps a `const TString&`?

Reply:
For the pure `const char*` case I found this: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rstr-zstring

Reply:
You mean some `fairroot::czstring`?

Because using gsl would mean introducing gsl as a public dependency. And as far as I remember, using it as a public dependency was not trivial.

Reply:
> You mean some fairroot::czstring?

yes, but perhaps out of scope of this PR. Let's discuss the pros and cons in one of the upcoming meetings?